pkg/resources/openstack/floatingip.go

```diff
@@ -55,14 +56,14 @@ func (os *clusterDiscoveryOS) listL3FloatingIPs(routerID string) ([]*resources.R
 	return resourceTrackers, nil
 }
 
-func (os *clusterDiscoveryOS) listFloatingIPs(instanceID string) ([]*resources.Resource, error) {
+func (os *clusterDiscoveryOS) listFloatingIPs(instance servers.Server) ([]*resources.Resource, error) {
 	var resourceTrackers []*resources.Resource
-	instance, err := os.osCloud.GetInstance(instanceID)
-	if err != nil {
-		return resourceTrackers, err
+	name := instance.Name
+	if val, ok := instance.Metadata[openstack.TagKopsName]; ok {
```

Comment:
If this is also used for backward compatibility, can you add a comment reflecting this?